### DISCUSSION TITLE
Weekly Update

### DISCUSSION THREAD

**Discussion by MikasaSucasa** [frc5567/2023-Robot#136] [Team Posts]
All,
Members reading this, please reach out to your fellow classmates and make sure they do the Test so that we are all communicating. I expect returners to find and help the newcomers.

I will be at practice tomorrow and we will begin to separate ourselves out. That being said, due to our size and immense amount of tasks (yes, there's A TON of things to do), we will begin to separate our large group into teams. These will be a mix of what everyone wants to do and what would be best for our team as whole. The latter is generally how the world works but I want everyone to enjoy themselves (within reason ;P ). 
Returners will be heading to these new groups almost immediately and the newcomers are going to begin coding our robot's drive train and simple controller operation (assuming Chris didn't already start this with you all hahaha).

Possible Sub-teams
1) **Electrical** (this will be headed by Brian and will include build team members. This will be a team that specializes in wiring our robot including documenting all sensors location and requirements. This relates directly with inputs into our software and this will not be an everyday thing)
2) **Vision** (this team will focus on autonomous operation and assistance programs utilizing cameras, color sensors, and others. This should be very dependent on all groups and should have a very good overall understanding of most areas.)
3-4) **Drive and Lift Controls** (this team will focus on the operation controls of the rest of the robot. For example, these groups will define speed for our robots drive train and lifts under different operating conditions. This means that this group will be doing math and logic to determine safe and logical operations using data from other sensors and requests from the vision team.)

All teams are equally important and tentative (may split you all up more/less). Please think about what sounds interesting to you. You all are expected to learn all of these areas but you will not be expected to necessarily be the group expert.

**Comment by coverbeek** [comment]
I created a couple sub-teams of the primary team to facilitate discussion. I will try to monitor all of the discussions and make sure there are answers to questions asked -- please don't be shy!!

Also, I put this link in the newcomers section, but it is useful for everyone -- the 2019 FRC Control System: 
https://wpilib.screenstepslive.com/s/currentCS/m/java

**Comment by MikasaSucasa** [comment]
If you have already downloaded VS code **COUGH CHRIS COUGH**, you will need to re-download the WPI version WITH a fresh VS Code install as the WPILib installer doesn't install in pre-existing installations. (Install is in that sentence a lot lewl)

**Comment by coverbeek** [comment]
Yep. I discovered the same. Everyone needs to install using the instructions on the 2019 FRC Control System page.
